### PR TITLE
Jeffrey branch

### PR DESCRIPTION
**Pull request by jeffreylin-21**
Added my half of the methods.

### CODE REVIEW

File: src/gr11review/part2/Main.java [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
+public static void main(String[] args) throws IOException {
+    
+    System.out.println(Utility.sumNumbers("12a1203b2"));
+    System.out.println(Utility.alphaWord("file"));
+    int[] num = {3, 2, 2, 3, 2, 0, 2};
+    num = Utility.notAlone(num, 2);
+    System.out.println(Arrays.toString(num));
+    int[] num2 = {2, 1, 1, 1, 2, 1};
+    System.out.println(Utility.canBalance(num2));
+    Utility.diagonal(8);
```
Comment (samliu21) [inline src/gr11review/part2/Main.java]: Make sure you delete this before handing it in

File: src/gr11review/part2/Utility.java [hunk 2]
```diff
@@ -1,5 +1,188 @@
...
+  // Takes in a string and returns the sum of the numbers within the string
+  public static int sumNumbers(String str) {
+
+    // Variable declaration and initialization
+    String strSum;
+    int intTotal, intLength;
+    char chrCharacter;
+    strSum = "";
+    intTotal = 0;
+    intLength = str.length();
```
Comment (samliu21) [inline src/gr11review/part2/Utility.java:27]: Maybe it's unnecessary to have a separate intLength variable?

File: src/gr11review/part2/Utility.java [hunk 3]
```diff
@@ -1,5 +1,188 @@
...
+      else if (!strSum.equals("")) {
+        intTotal += Integer.parseInt(strSum);
+        strSum = "";
+      }
+    }
+    // Checks if there was a number at the end of the string and adds it to the total
+    if (!strSum.equals("")) {
+      intTotal += Integer.parseInt(strSum);
+    } 
+    // Returns the total
```
Comment (samliu21) [inline src/gr11review/part2/Utility.java:47]: You should probably put a space before the "return total" section

File: src/gr11review/part2/Utility.java [hunk 4]
```diff
@@ -1,5 +1,188 @@
...
+    }
     
+    // Close the file and return the string
+    theFile.close();
+    return strAlpha;
+
+  }
+
+  // JEFFREY
+  // Returns an array where all instances of the selected value in the array are turned into its numerically highest neighbor
```
Comment (samliu21) [inline src/gr11review/part2/Utility.java]: I think "numerically greater neighbor" fits better here

File: src/gr11review/part2/Utility.java [hunk 5]
```diff
@@ -1,5 +1,188 @@
...
+    // Variable initialization and declaration
+    int intLength;
+    intLength = nums.length;
+
+    // Iterates through the array
+    for (int intCount = 1; intCount < intLength-1; intCount++) {
+      
+      // If the current number equals value and doesn't equal the numbers before and after it, set the current number equal to the max of the two numbers adjacent to it
+      if (nums[intCount] == value) {
+        if (nums[intCount-1] != nums[intCount] && nums[intCount+1] != nums[intCount]) {
```
Comment (samliu21) [inline src/gr11review/part2/Utility.java]: Put a space before and after minus and plus signs

File: src/gr11review/part2/Utility.java [hunk 6]
```diff
@@ -1,5 +1,188 @@
...
+    int intLength;
+    int intSum;
+    int intPreSum;
+    intLength = nums.length;
+    intSum = 0;
+    intPreSum = 0;
+
+    // Obtain the sum of all numbers in the array 
+    for (int intCount = 0; intCount < intLength; intCount++) {
+
```
Comment (samliu21) [inline src/gr11review/part2/Utility.java]: Delete spaces in between

**Review by samliu21 — CHANGES_REQUESTED** [review]
The functionality of the code looks okay. A few formatting changes here and there.

**Comment by jeffreylin-21** [comment]
@samliu1234 take another look please.

**Pull request merged**